baseapp/options.go

```diff
@@ -122,9 +124,22 @@ func (app *BaseApp) SetVersion(v string) {
 	app.version = v
 }
 
-// SetProtocolVersion sets the application's protocol version
-func (app *BaseApp) SetProtocolVersion(v uint64) {
-	app.appVersion = v
+// SetAppVersion sets the application's version this is used as part of the
+// header in blocks and is returned to the consensus engine in EndBlock.
+func (app *BaseApp) SetAppVersion(ctx context.Context, v uint64) error {
+	if app.paramStore == nil {
+		return errors.New("param store must be set to set app version")
+	}
+
+	cp, err := app.paramStore.Get(ctx)
+	if err != nil {
+		return err
+	}
+	cp.Version.App = v
```

Comment:
considering https://github.com/cosmos/cosmos-sdk/pull/16244/files#diff-328a55b5ac5816102544792c5c6a75df50df3b9b01afe770c9abc2da79b3e6afR208
can we check `cp.Version == nil` here as well?